src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs

```diff
@@ -359,6 +359,7 @@ private static async ValueTask ReadToFillAsync(Stream stream, Memory<byte> buffe
 
             if (bytesRead < buffer.Length)
             {
+                // TODO: How should we categorize this? Seems more like a connection establishment than than InvalidResponse
```

Comment:
How would errors during the establishment of a CONNECT proxy tunnel be surfaced (e.g. proxy sent a garbled response -- not just a valid but non-200 one)?

`ProxyTunnelError`'s description also seems to fit this case (we got a connection but failed to agree with the proxy).

Reply:
Added a new message for the topmost `HttpRequestException` and decided to leave inner SocksExceptions as they are.